docs: enable mTLS with the mqtt-proxy plugin (#7435)
apache/apisix: docs/en/latest/plugins/mqtt-proxy.md

MQTT connections with different client ID will be forwarded to different nodes based on the consistent hash algorithm. If client ID is missing, client IP is used instead for load balancing.

## Enabling mTLS with mqtt-proxy plugin

> Stream proxies use TCP connections and can accept TLS. Follow the guide about [how to accept tls over tcp connections](https://apisix.apache.org/docs/apisix/stream-proxy/#accept-tls-over-tcp-connection) to open a stream proxy with enabled TLS.

Reviewer: ```suggestion
Stream proxies use TCP connections and can accept TLS. Follow the guide about [how to accept tls over tcp connections](../stream-proxy.md/#accept-tls-over-tcp-connection) to open a stream proxy with enabled TLS.
```

The `mqtt-proxy` plugin is enabled through TCP communications on the specified port for the stream proxy, and will also require clients to authenticate via TLS if tls is set to true.

> Configure `ssl` providing the CA certificate and the server certificate, together with a list of SNIs. Steps to protect `stream_routes` with `ssl` are equivalent to the ones to [protect Routes](https://apisix.apache.org/docs/apisix/mtls/#protect-route)

Reviewer: ```suggestion
Configure `ssl` providing the CA certificate and the server certificate, together with a list of SNIs. Steps to protect `stream_routes` with `ssl` are equivalent to the ones to [protect Routes](../mtls.md/#protect-route)
```

### Create a stream_route using mqtt-proxy plugin and mTLS

Here is an example of how create a stream_route which is using the `mqtt-proxy` plugin, providing the CA certificate, the client certificate and the client key (for self-signed certificates which are not trusted by your host, use the `-k` flag):

> ```

Reviewer 2: ```suggestion
```shell
```

> curl --cacert /path/to/ca.pem --key /path/to/client.key.pem --cert /path/to/client.pem  127.0.0.1:9180/apisix/admin/stream_routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '

Reviewer 3: IMO, here use `--cacert` `--cert ` `--key` in `curl` means that the APISIX adminAPI was enabled mTLS, not the stream route enabled mTLS.

Author: Actually it is. Both mTLS  for the admin API and the mqtt proxy were enabled. I think this flags sould be removed from the curl request to avoid confusion.

Reviewer 3: + 1 for remove this flag

{
    "plugins": {
        "mqtt-proxy": {
            "protocol_name": "MQTT",
            "protocol_level": 4
        }
    },

>     "sni": "<your_sni_name>",

Reviewer: ```suggestion
    "sni": "${your_sni_name}",
```

    "upstream": {
        "nodes": {

>             "<your_upstream_node_name>": 1

Reviewer: ```suggestion
            "127.0.0.1:1980": 1
```

Reviewer: My suggestion is to use 127.0.0.1, which is unified with other documentation.

        },
        "type": "roundrobin"
    }
}'
```

> The sni name must math one or more of the SNIs provided to the ssl object that you created with the CA and server certificates.

Reviewer 4: ```suggestion
The `sni` name must math one or more of the SNIs provided to the SSL object that you created with the CA and server certificates.
```
